App/xcbuddykit/Sources/Generator/ConfigGenerator.swift

```diff
@@ -66,61 +71,67 @@ final class ConfigGenerator: ConfigGenerating {
     ///   - groups: Project groups.
     ///   - sourceRootPath: path to the folder that contains the generated project.
     ///   - context: generation context.
+    ///   - options: generation options.
     /// - Returns: the confniguration list reference.
     /// - Throws: an error if the generation fails.
     func generateProjectConfig(project: Project,
                                pbxproj: PBXProj,
                                groups: ProjectGroups,
                                sourceRootPath: AbsolutePath,
-                               context: GeneratorContexting) throws -> PBXObjectReference {
+                               context: GeneratorContexting,
+                               options: GenerationOptions) throws -> PBXObjectReference {
         /// Default build settings
         let defaultAll = BuildSettingsProvider.projectDefault(variant: .all)
         let defaultDebug = BuildSettingsProvider.projectDefault(variant: .debug)
         let defaultRelease = BuildSettingsProvider.projectDefault(variant: .release)
 
+        /// Configuration list
+        let configurationList = XCConfigurationList(buildConfigurations: [])
+        let configurationListReference = pbxproj.objects.addObject(configurationList)
+
         // Debug
-        var debugSettings: [String: Any] = [:]
-        extend(buildSettings: &debugSettings, with: defaultAll)
-        extend(buildSettings: &debugSettings, with: project.settings?.base ?? [:])
-        extend(buildSettings: &debugSettings, with: defaultDebug)
-        let debugConfiguration = XCBuildConfiguration(name: "Debug")
-        if let debugConfig = project.settings?.debug {
-            extend(buildSettings: &debugSettings, with: debugConfig.settings)
-            if let xcconfigDebug = debugConfig.xcconfig {
-                let fileReference = try fileGenerator.generateFile(path: xcconfigDebug,
-                                                                   in: groups.projectConfigurations(),
-                                                                   sourceRootPath: sourceRootPath,
-                                                                   context: context)
-                debugConfiguration.baseConfigurationReference = fileReference.reference
+        if options.buildConfiguration == .debug {
+            var debugSettings: [String: Any] = [:]
```

Comment:
I think it would make sense to extract this `L:94-111` to a helper function that receives the `GenerationOptions` value and then generates the configuration and returns it.

If we do it like this we can support no only `.debug` & `.release` but eventually something like `.custom("beta")`.

we could even receive a list of configurations and iterate through them and generate all the ones in one go

Reply:
Just to clarify; it can receive the `GenerationOptions` or just the `BuildConfiguration` (since this is the one it would really need); but the idea is still the same